### PR TITLE
#161967122 document app with swagger

### PR DESCRIPTION
**Pull request by denomash**
**What does this PR do?**
- add documentation to Authors Haven

**Description of Task to be completed?**
- document app using swagger

**How should this be manually tested?**

- Clone the repo and cd into the project folder
- Create a virtual environment by running `python3 -m venv <name_of_environment>`
- Activate the virtual environment by running `source <name_of_environment>/bin/activate`
- Run `$ pip install -r requirements.txt`
- Then add the following environment variables to your computer.

 - `$ export DB_NAME='choose_a_dbname'`
 - `$ export DB_USER='db_username'`
 - `$ export DB_PASSWORD='db_role_password'`
 - `$ export DB_HOST='localhost or preffered host'`
 - `$ export DB_PORT='server_port_number'`
- `$ export SECRET_KEY='some_secret_key`

- Change the current branch by running `git checkout ft-document-app-with-swagger-161967122`
- Run `python3 manage.py runserver`
- open your favorite browser and use this URL `http://127.0.0.1:8000`

**What are the relevant pivotal tracker stories?**
- [#161967122](https://www.pivotaltracker.com/story/show/161967122/comments/197223584)

**Checklist (edited)**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

**Screenshots (if appropriate)**

screenshot 2018-12-04 at 12 16 52

screenshot 2018-12-04 at 12 26 43

screenshot 2018-12-04 at 12 26 58

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,6 +24,7 @@ var/
 .installed.cfg
 *.egg
 .idea/
+*.DS_Store
```
Comment (bettblake08) [inline .gitignore:27]: Remove duplicate entries of DS_Store

**Review by bettblake08 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (denomash) [reply .gitignore:27]: ok on it

**Review by bettblake08 — APPROVED** [review]

File: authors/settings.py [hunk 2]
```diff
@@ -15,21 +15,20 @@
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
-
-
 # Quick-start development settings - unsuitable for production
 # See https://docs.djangoproject.com/en/1.11/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
 SECRET_KEY = os.getenv('SECRET_KEY')
```
Comment (jillo-abdullahi) [inline authors/settings.py:22]: @denomash The SECRET_KEY is set from the environment. Please add an export for it under the test instructions on this PR. Also I did not need to add the other DB exports to view the documentation. I only had to do a migrate.

File: authors/apps/authentication/views.py [hunk 3]
```diff
@@ -2,15 +2,15 @@
 from rest_framework.generics import RetrieveUpdateAPIView
 from rest_framework.permissions import AllowAny, IsAuthenticated
 from rest_framework.response import Response
-from rest_framework.views import APIView
+from rest_framework.generics import CreateAPIView
```
Comment (jillo-abdullahi) [inline authors/apps/authentication/views.py:5]: @denomash Any reason for us to switch to generic views?

**Review by jillo-abdullahi — COMMENTED** [review]
@denomash There is a /docs/api endpoint visible on the documentation and when you execute, it gives a 406, undocumented, error. Should it be visible on the documentation?

File: authors/settings.py (hunk 2 above)
Comment (denomash) [reply authors/settings.py:22]: Nice observation @jillo-abdullahi. Let me add that in the PR

File: authors/apps/authentication/views.py (hunk 3 above)
Comment (denomash) [reply authors/apps/authentication/views.py:5]: Swagger best works with generics, unlike views.

**Comment by Darius-Ndubi** [comment]
On the screenshot, the payload is not formatted how its shows in the screenshot up there, what could be the reason?

**Review by mutoromike — CHANGES_REQUESTED** [review]
Update the Readme file with a link to the documentation.

**Comment by denomash** [comment]
@mutoromike the documentation works only locally for now since this branch is not deployed to Heroku. The documentation will work after merging to develop.

File: authors/settings.py (hunk 2 above)
Comment (jillo-abdullahi) [reply authors/settings.py:22]: Awesome!

File: authors/apps/authentication/views.py (hunk 3 above)
Comment (jillo-abdullahi) [reply authors/apps/authentication/views.py:5]: Noted.

**Review by Darius-Ndubi — APPROVED** [review]
👍

**Review by Bryan-Cee — APPROVED** [review]
👍

**Review by mutoromike — APPROVED** [review]

**Comment by mutoromike** [comment]
👍

**Pull request merged**